micronota/parsers/cmscan.py

```diff
@@ -0,0 +1,413 @@
+r'''
+CMSCAN Parser
+=============
+
+The program cmscan of the Infernal suite [1]_ is a tools to screen a nucleotide
+sequence for homology to known functional RNA families. Input is a set of fasta
+sequences and a set of family models. Output is a set of potential hits of sub-
+sequences to one of the family models.
+
+Output of cmscan comes in two flavors. STDOUT provides complete hit
+reports, but is hard to be parsed. If the flag --tbloutput FILENAME is used,
+FILENAME "save parseable table of hits to file". (Alignment between query
+sequence and family model are **not** included.) This python module is for
+parsing the tbloutput format.
+
+This example lists the first hits between the chromosome of E.Coli and Rfam 12
+using CMscan:
+
+.. code-block:: none
+#target name           accession query name                   accession mdl mdl from   mdl to seq from   seq to strand trunc pass   gc  bias  score   E-value inc description of target
+#--------------------- --------- ---------------------------- --------- --- -------- -------- -------- -------- ------ ----- ---- ---- ----- ------ --------- --- ---------------------
+LSU_rRNA_bacteria      RF02541   gi|15829254|ref|NC_002695.1| -          cm        1     2925  4977823  4980727      +    no    1 0.53  45.4 2889.8         0 !   -
+LSU_rRNA_bacteria      RF02541   gi|15829254|ref|NC_002695.1| -          cm        1     2925  5018849  5021753      +    no    1 0.53  45.4 2889.8         0 !   -
+LSU_rRNA_bacteria      RF02541   gi|15829254|ref|NC_002695.1| -          cm        1     2925  4737148  4740052      +    no    1 0.53  45.4 2889.5         0 !   -
+LSU_rRNA_bacteria      RF02541   gi|15829254|ref|NC_002695.1| -          cm        1     2925  4833642  4836546      +    no    1 0.53  45.4 2889.5         0 !   -
+LSU_rRNA_bacteria      RF02541   gi|15829254|ref|NC_002695.1| -          cm        1     2925   229090   231992      +    no    1 0.53  43.8 2888.4         0 !   -
+LSU_rRNA_archaea       RF02540   gi|15829254|ref|NC_002695.1| -          cm        1     2990  4977822  4980726      +    no    1 0.53  45.8 1848.7         0 !   -
```

Comment:
For doc-tests, its good practice to also follow numpy-doc styles.

Here's an example of how we implement this in [skbio](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/composition.py#L747)